shithead_game_server/src/lobby.rs

```diff
@@ -29,10 +26,13 @@ impl CardsById {
     pub fn new() -> Self {
         let mut cards_by_id = Vec::new();
 
-        // this part is just for testing purposes
-        // in the future when the card struct is fully implemented this will initialize an actual deck
-        for _ in 0..54 {
-            cards_by_id.push(Card { rank: (), suit: () })
+        // initialize with all possible cards
+        for rank in Rank::iter() {
+            // if the card is a joker use only 2 suits, otherwise use all suits
+            let suits_amount = if rank == Rank::Joker { 2 } else { Suit::COUNT };
```

Comment:
You're using an integer for the joker amount and then using the count of suits for the other cards. It's a little bit weird but ok